core/src/components/datetime/datetime.tsx

```diff
@@ -1259,6 +1233,67 @@ export class Datetime implements ComponentInterface {
        */
       this.activeParts = [];
     }
+
+    /**
+     * Only animate if:
+     * 1. We're using grid style (wheel style pickers should just jump to new value)
+     * 2. The month and/or year actually changed, and both are defined (otherwise there's nothing to animate to)
+     * 3. The calendar body is visible (prevents animation when in collapsed datetime-button, for example)
+     * 4. The month/year picker is not open (since you wouldn't see the animation anyway)
+     */
+    const didChangeMonth = month !== workingParts.month || year !== workingParts.year;
+    const monthYearDefined = month !== undefined && year !== undefined;
```

Comment:
Nit: I think these could be combined since we can only really consider a month/year to have changed if the month/year is defined in the first place
```suggestion
    const didChangeMonth = (month !== undefined && month !== workingParts.month) || (year !== undefined && year !== workingParts.year);
```